src/test/fuzz/fuzz.cpp

```diff
@@ -79,6 +79,26 @@ void FuzzFrameworkRegisterTarget(std::string_view name, TypeTestOneInput target,
 static std::string_view g_fuzz_target;
 static const TypeTestOneInput* g_test_one_input{nullptr};
 
+
+#if defined(__clang__) && defined(__linux__)
```

Comment:
I don't know how to make this work outside of these params. I don't want to use a macro, I think this should always be enabled if possible.

Reply:
https://cirrus-ci.com/task/6736987506343936:

```
Undefined symbols for architecture x86_64:
  "___llvm_profile_reset_counters", referenced from:
      initialize() in libtest_fuzz.a(libtest_fuzz_a-fuzz.o)
  "___gcov_reset", referenced from:
      initialize() in libtest_fuzz.a(libtest_fuzz_a-fuzz.o)
ld: symbol(s) not found for architecture x86_64
```

https://cirrus-ci.com/task/6596250017988608:

```
/usr/bin/ld: libtest_fuzz.a(libtest_fuzz_a-fuzz.o): in function `ResetCoverageCounters()':
./src/test/fuzz/fuzz.cpp:90: undefined reference to `__llvm_profile_reset_counters'
/usr/bin/ld: ./src/test/fuzz/fuzz.cpp:94: undefined reference to `__gcov_reset'
/usr/bin/ld: ./src/test/fuzz/fuzz.cpp:90: undefined reference to `__llvm_profile_reset_counters'
/usr/bin/ld: ./src/test/fuzz/fuzz.cpp:94: undefined reference to `__gcov_reset'
collect2: error: ld returned 1 exit status
```

https://cirrus-ci.com/task/5470350111145984:

```
test/fuzz/fuzz.cpp:84:1: error: attributes are not permitted in this position [-Werror=attributes]
   84 | __attribute__((weak)) extern "C" void __llvm_profile_reset_counters(void);
      | ^~~~~~~~~~~~~
test/fuzz/fuzz.cpp:84:33: note: attributes may be inserted here
   84 | __attribute__((weak)) extern "C" void __llvm_profile_reset_counters(void);
      |                                 ^
test/fuzz/fuzz.cpp:85:1: error: attributes are not permitted in this position [-Werror=attributes]
   85 | __attribute__((weak)) extern "C" void __gcov_reset(void);
      | ^~~~~~~~~~~~~
test/fuzz/fuzz.cpp:85:33: note: attributes may be inserted here
   85 | __attribute__((weak)) extern "C" void __gcov_reset(void);
      |                                 ^
test/fuzz/fuzz.cpp: In function ‘void ResetCoverageCounters()’:
test/fuzz/fuzz.cpp:89:9: error: the address of ‘void __llvm_profile_reset_counters()’ will never be NULL [-Werror=address]
   89 |     if (__llvm_profile_reset_counters) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test/fuzz/fuzz.cpp:84:39: note: ‘void __llvm_profile_reset_counters()’ declared here
   84 | __attribute__((weak)) extern "C" void __llvm_profile_reset_counters(void);
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test/fuzz/fuzz.cpp:93:9: error: the address of ‘void __gcov_reset()’ will never be NULL [-Werror=address]
   93 |     if (__gcov_reset) {
      |         ^~~~~~~~~~~~
test/fuzz/fuzz.cpp:85:39: note: ‘void __gcov_reset()’ declared here
   85 | __attribute__((weak)) extern "C" void __gcov_reset(void);
      |                                       ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

Reply:
Might followup here.